### PR TITLE
Introduce manual mapping of some ILP64 routines

### PR DESCRIPTION
**Pull request by imciner2**
MKL doesn't export all routines over the ILP64 interface in libmkl_rt, specifically it doesn't export functions that might not be affected by the differing integer sizes (e.g. no integers used in the API).

In these cases, we can redirect the ILP64 version to the LP64 version to maintain compatibility.

This list is not complete, but is basically what I found scrolling through the lapack header provided with MKL. We should ideally find a way to automate finding these functions and script it.

First noticed in #138.

### CODE REVIEW

**Comment by staticfloat** [comment]
This is quite annoying.  Should we perhaps maintain a list in LBT and automatically do this mapping if these functions are not found in an ILP64 BLAS, but an LP64 BLAS happens to be loaded?  This would then create a functional difference between the order of loading of the two interfaces, but I think it could be a sane default for users.

**Comment by ViralBShah** [comment]
We should report this to Intel as well.

**Comment by ViralBShah** [comment]
@staticfloat Should we merge this PR for now until there is a new upstream MKL release?

**Comment by imciner2** [comment]
> Should we perhaps maintain a list in LBT and automatically do this mapping if these functions are not found in an ILP64 BLAS, but an LP64 BLAS happens to be loaded? This would then create a functional difference between the order of loading of the two interfaces, but I think it could be a sane default for users.

If this affects multiple BLAS/LAPACK wrappers, it might be beneficial to do this in LBT proper instead of the wrappers here. I guess there is a theory it also affects BLIS according to the link issue? I guess the best way to do this is to build a complete API list of functions that don't have integers in their API and keep that in LBT. Then on load, we can remap the ILP64 version to the LP64 version if it still has the default function after all other symbols are loaded. One thing I noticed when looking at the API was that some functions that returned integers also didn't have an ILP64 variant (when their arguments were also only double/float), since I guess the extra sizes weren't needed for the return value.

Creating that list will be annoying, though. It took me about 1.5 hours just to pull the current list out, and that is just basically half the MKL lapack header (I didn't even touch lapacke or cblas).

In short, it is a bit of a mess trying to identify all these functions, and it would be really nice if the upstream implementations could just export the symbols (even if it is their own trampoline to the LP64 variant they already have).

**Comment by amontoison** [comment]
I think that we should add this mapping between LP64 / ILP64 in the LBT repository. It will make LBT more robust to different BLAS / LAPACK backends.
Creating the list is annoying but we just need to do thzt once.

**Comment by mkrainiuk** [comment]
> @mkrainiuk It would be nice for us to have all the ILP64 functions exported with the right suffixes.

We're planning these changes for the next release.

**Comment by mkrainiuk** [comment]
>  @mkrainiuk do you know if the changes made it in time for 2024.0.0, or will it be a later release?

Unfortunately, it was too late for 2024.0, but it should be part of the next update release.

**Comment by ViralBShah** [comment]
@imciner2 Should we merge this in the meanwhile? Is x86 windows failure something problematic (I haven't checked the logs)?

**Comment by ViralBShah** [comment]
Tagged 0.6.2: https://github.com/JuliaRegistries/General/pull/98215

**Comment by ViralBShah** [comment]
I believe this is now fixed in MKL 2024.1.
